shared/src/crypto/signature.rs

```diff
@@ -30,6 +30,8 @@ pub const SECP_PUB_LEN: usize = 65;
 pub enum SignatureType {
     Secp256k1 = 1,
     BLS = 2,
+    // TODO, should this be a different number?
+    External = 256,
```

Comment:
This is not correct. I think this type is used only in the `validate_signature` syscall. And there is no such thing as validating an external signature. Instead, the signed chain message needs to specify "Delegated" as the signature type at the client level:

- https://github.com/filecoin-project/lotus/blob/aa0aaccc243774763a27db0a694f8014f1e3ea53/chain/types/signedmessage.go#L45-L48
- https://github.com/filecoin-project/go-state-types/blob/master/crypto/signature.go#L14-L19

This would prompt the FVM to delegate the signature to the validate entrypoint in the sender actor.

Reply:
`Delegated` is a much better signature type name. There was some confusion about signature validation method calls (deal signatures, which is validated by `validate_signature`  _is not_ message validation that happens in lotus etc.)